applications/DEMApplication/python_scripts/analytic_tools/analytic_data_procedures.py

```diff
@@ -127,32 +160,38 @@ def UpdateDataBases(self, time):
         else:
             self.CreateDataFile(time)
 
+
+
     def UpdateDataFile(self, time):
 
         # how to create subgrouped datasets with variable name:
         # group2 = f.create_group('group2/subfolder')
         # group2.create_dataset('data',data=d)
         import h5py
         with h5py.File(self.new_path, 'a') as f, h5py.File(self.old_path, 'r') as f_old:
-            for sub_part in self.model_part:
-                if sub_part[IS_GHOST]:
-                    shape, time, n_particles, mass, avg_vel_nr = self.face_watcher_analysers[sub_part.Name].UpdateData(time)
-                    shape_old = f_old['/' + sub_part.Name + '/time'].shape
+            #for sub_part in self.sub_model_parts:
+            for i in range(self.surface_analyzers_list_size):
+                #if sub_part[IS_GHOST]:
+                if self.surface_analyzers_list[i].ghost:
```

Comment:
it should be always True, not necessary, right?

Reply:
because:
```
for smp in sub_model_parts:
            if smp[IS_GHOST] == True:
                self.ghost_smp_detected = True
                self.surface_analyzers_list.append(SurfaceAnalyzer(smp))
```